danascheider/skyrim_inventory_management_frontend
# Redesign flash messages

## Context

[**Redesign flash messages**](https://trello.com/c/mi1ybGp3/116-redesign-flash-messages)

The `FlashMessage` component is causing a lot of issues with reflow when it is shown and hidden. Additionally, it's kind of ugly. We needed to redesign the flash message into a fixed-positioned element that would be less obtrusive.

## Changes

* Redesign flash message component to be a fixed-position element
* Hide flash message after 4 seconds when it is shown
* Display flash message on shopping list creation (this was something I'd been meaning to change for a while)
* Change mechanism by which flash is shown or hidden (conditionally adding a class rather than conditionally rendering the component)
* Create `LoadingError` component to be shown instead of flash if games or shopping lists fail to load

## Considerations

I didn't like using the new flash component to display loading errors for games and shopping lists, especially since it now disappears after 4 seconds. So, I designed a new component to be rendered for the error loading state.

These changes have been tested and work well in Safari.

## Manual Test Cases

Walk through any case where a flash message would be shown and make sure it still works and looks good with the new design/disappearing after 4 seconds. Make sure any code that calls `setFlashVisible` runs to make sure everything looks right.

## Screenshots and GIFs

### Short Message, 1201px

![issue-116_1201px](https://user-images.githubusercontent.com/5115928/128434168-8fdcb65e-d2d2-4e2e-a0ba-2499ecbae475.png)

### Short Message, 1025px

![issue-116_1025px](https://user-images.githubusercontent.com/5115928/128434178-60264da9-aaa7-4fea-b2c3-defd8963a171.png)

### Short Message, 769px

![issue-116_769px](https://user-images.githubusercontent.com/5115928/128434186-c4dd2011-5e60-4e7e-bbcb-b5ef7d7c4529.png)

### Short Message, 601px

![issue-116_601px](https://user-images.githubusercontent.com/5115928/128434205-3fa4d0e2-fa59-485d-8f65-73172c5fdebd.png)

### Short Message, 425px

![issue-116_425px](https://user-images.githubusercontent.com/5115928/128434219-1b707b89-7cc5-4334-95c2-a2827bfcfffe.png)

### Long Message, 425px

![issue-116_long-message_425px](https://user-images.githubusercontent.com/5115928/128434246-ab2bebd1-bf6e-423e-a768-5e911958fce3.png)

### Loading Error Component, 1201px

<img width="1169" alt="Screen Shot 2021-08-06 at 10 25 31 am" src="https://user-images.githubusercontent.com/5115928/128437869-bdcc4c6e-807c-48d9-a4a8-11379cfc52ea.png">

### Loading Error Component, 1025px

<img width="992" alt="Screen Shot 2021-08-06 at 10 25 44 am" src="https://user-images.githubusercontent.com/5115928/128437886-1fd6a9ee-cad6-4d59-9b13-a94f652f0606.png">

### Loading Error Component, 769px

<img width="737" alt="Screen Shot 2021-08-06 at 10 25 56 am" src="https://user-images.githubusercontent.com/5115928/128437901-36a35365-6ce6-441d-b0a9-9897fd0216e6.png">

### Loading Error Component, 601px

<img width="568" alt="Screen Shot 2021-08-06 at 10 26 07 am" src="https://user-images.githubusercontent.com/5115928/128437915-bf73c83f-c8a5-4588-a260-669162737612.png">

### Loading Error Component, 425px

<img width="390" alt="Screen Shot 2021-08-06 at 10 26 22 am" src="https://user-images.githubusercontent.com/5115928/128437932-e46ce4a5-3e30-41ac-8664-f55dabb4ee98.png">
